File: day_3/p1.py

def read_file():
	with open("input", "r") as f:
		full_file = f.read()
	return full_file.split("\n")
	

def main():
	score = 0
	read_list = read_file()
	for item in read_list:
		a = item[:int(len(item)/2)]
		b = item[int(len(item)/2):]
		# for char in a:
		# 	if 
		match = [char for char in a if char in b][0]
		print(match)
		# print(ord(str(match))-96)
		score_str = " abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
		print(score_str.index(match))
		score += score_str.index(match)
	print(score)



if __name__ == '__main__':
	main()

File: day_3/p1.rs
use std::fs::File;
use std::io::{BufRead, BufReader};

fn main() {
    let filename = "input";
    let file = File::open(filename).unwrap();
    let reader = BufReader::new(file);
    let mut score = 0;
    for (index, line) in reader.lines().enumerate() {

        let line = line.unwrap();
        let first_half = &line[0..line.len()/2];
        let last_half = &line[line.len()/2..];
        let abctrng = " abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
        let first_half_chars = first_half.chars();

        for c in first_half_chars {
            if last_half.contains(c){

                let char_val = abctrng.find(c);

                match char_val {
                    Some(val) => {
                        score += val;
                        println!("{}: {}", c, val);
                        println!("{}, {}|{}", index + 1, first_half, last_half);
                    }
                    None => {}
                }
                break;
            }
        }
    }
    println!("{}", score);
}

